### DISCUSSION TITLE
naming standards for modules

### DISCUSSION THREAD

**Discussion by strangemonad** [modular/modular#205] [Ideas]
If I'm coming from the python world, while upper-cased module names are allowed, they're certainly not common. More recently, python packaging has generally encouraged all lowercase with hyphens for distribution names and all lowercase with underscores for modules and packages.

The naming of modules in the mojo std lib feels more reminiscent of Swift and Core Foundation. 

I know the packaging and distribution story hasn't been fully sorted out yet. I'd hope that mojo, being a superset of python, might consider supporting publishing native binaries to pypi. This would be in line with other libraries that produce native bindings e.g. cython or rust-pyo3

**Comment by ninyawee** [comment]
I have 10+ opensource pypi libs. see the point. I agreed with the observation.
personally, I'm ok with the both. The titlecase, it is make language a bit tidy.
the snakecase makes it feels home.

**Reply by strangemonad** [reply]
exactly. As a polyglot, I'm comfortable with both. While python allows both, when I see snake_case for modules and hyphen-case for distributions, I feel like I'm in modern python
